### PR TITLE
Add export job xml feature

### PR DESCRIPTION
**Pull request by amouhoub**
In this PR, we add the Export JOB's XML feature to the portal.
The PR contains code related to: Adding btn, Adding API methods, Adding DownloadServlet & Refactoring some legacy code

### CODE REVIEW

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 1]
```diff
@@ -70,6 +70,9 @@
     @Source("images/job_kill_16.png")
     ImageResource job_kill_16();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 2]
```diff
@@ -70,6 +70,9 @@
...
 
+    @Source("images/job_export_32.png")
+    ImageResource scheduler_export_32();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 3]
```diff
@@ -70,6 +70,9 @@
...
+
     @Source("images/job_output_16.png")
     ImageResource job_output_16();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ */
+@SuppressWarnings("serial")
+public class DownloadJobXMLServlet extends HttpServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DownloadJobXMLServlet.class);
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        getJobXML(request, response);
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.scheduler.server;
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00120)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(DownloadJobXMLServlet.class);
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        getJobXML(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        getJobXML(request, response);
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java [hunk 7]
```diff
@@ -79,6 +81,8 @@
 
     private static Logger LOGGER = Logger.getLogger(JobsController.class.getName());
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java]: MINOR Rename this field "LOGGER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java [hunk 8]
```diff
@@ -770,6 +769,10 @@ public InputStream apply(RestClient restClient) {
         });
     }
 
+    public String getJobXML(final String sessionId, final String jobId) throws RestServerException, ServiceException {
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java [hunk 9]
```diff
@@ -236,37 +228,33 @@ private boolean isRowSelected(int rowNum) {
...
-                Job job = JobRecord.getJob(record);
-                String progressCounters = job.getFinishedTasks() + " / " + job.getTotalTasks();
-                return "<div style='" + style + "'>" + progressCounters + "</div>";
+        progressField.setCellFormatter((value, record, rowNum, colNum) -> {
+            int pw = getFieldWidth(PROGRESS_ATTR.getName());
+            float progress = 0;
+            if (value != null) {
+                progress = Float.parseFloat(value.toString());
             }
+            int bx = new Double(Math.ceil(pw * progress)).intValue() - 601;
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java]: MAJOR Remove this "Double" constructor [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
MINOR Remove the boxing to "Double". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2153)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 161 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 42 major
* ![MINOR][MINOR] 118 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [SchedulerImages.java#L39](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java#L39): Move constants to a class or enum. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1214)
1. ![MAJOR][MAJOR] [RestClient.java#L290](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L290): Method has 10 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RestClient.java#L322](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L322): Method has 11 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L192](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L192): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L235](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L235): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L272](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L272): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L368](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L368): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L386](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L386): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L396](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L396): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L407](https://github.com/ow2-proactive/scheduling-portal/blob/fedc6a3a9968c246060da768ef3d10b3d3abc652/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L407): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 10]
```diff
@@ -70,6 +71,9 @@
     @Source("images/job_kill_16.png")
     ImageResource job_kill_16();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 11]
```diff
@@ -70,6 +71,9 @@
...
 
+    @Source("images/job_export_32.png")
+    ImageResource scheduler_export_32();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 12]
```diff
@@ -70,6 +71,9 @@
...
+
     @Source("images/job_output_16.png")
     ImageResource job_output_16();
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 4 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 5 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00120)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 6 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java (hunk 7 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java]: MINOR Rename this field "LOGGER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java (hunk 8 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java (hunk 9 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java]: MAJOR Remove this "Double" constructor [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
MINOR Remove the boxing to "Double". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2153)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 160 issues
* ![MAJOR][MAJOR] 42 major
* ![MINOR][MINOR] 118 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RestClient.java#L290](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L290): Method has 10 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RestClient.java#L322](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L322): Method has 11 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L192](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L192): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L235](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L235): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L272](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L272): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L368](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L368): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L386](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L386): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L396](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L396): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L407](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L407): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L418](https://github.com/ow2-proactive/scheduling-portal/blob/889be688b5d415143316c420bb0565e5c4478fab/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L418): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java [hunk 13]
```diff
@@ -36,6 +36,7 @@
  * @author mschnoor
  *
  */
+@SuppressWarnings("squid:S1214")
```
Comment (Mykhailenko) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: Could you please put comment why do you suppress this warning?

Comment (amouhoub) [reply scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: @Mykhailenko I need to refactor this class, which I'm doing right away, but I need to do it in a separate PR, therefore I didn't want to put this personal reason in a comment. (TODO later)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java (hunk 10 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java (hunk 11 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java (hunk 12 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerImages.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 4 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 5 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00120)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java (hunk 6 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java]: MINOR Add a "try/catch" block for "getJobXML". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java [hunk 14]
```diff
@@ -79,6 +83,8 @@
...
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java]: MINOR Rename this field "LOGGER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java (hunk 8 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java (hunk 9 above)
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java]: MAJOR Remove this "Double" constructor [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
MINOR Remove the boxing to "Double". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2153)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 160 issues
* ![MAJOR][MAJOR] 42 major
* ![MINOR][MINOR] 118 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RestClient.java#L290](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L290): Method has 10 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RestClient.java#L322](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L322): Method has 11 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L192](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L192): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L235](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L235): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L272](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L272): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L368](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L368): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L386](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L386): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L396](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L396): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L407](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L407): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L418](https://github.com/ow2-proactive/scheduling-portal/blob/e5dbdf17c9bf763387e25712f63853f32d462201/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L418): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**